llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp

```diff
@@ -1425,8 +1453,14 @@ static bool runImpl(Module &M, AnalysisGetter &AG, TargetMachine &TM,
     }
   }
 
-  ChangeStatus Change = A.run();
-  return Change == ChangeStatus::CHANGED;
+  bool Changed = A.run() == ChangeStatus::CHANGED;
+
+  if (Changed && (LTOPhase == ThinOrFullLTOPhase::None ||
+                  LTOPhase == ThinOrFullLTOPhase::FullLTOPostLink ||
+                  LTOPhase == ThinOrFullLTOPhase::ThinLTOPostLink))
+    checkWavesPerEU(M, TM);
```

Comment:
This isn't running as part of the attribute run? Why isn't this part of manifest? 

Changed is also really course, you only can care about specific attributes changing

Reply:
> This isn't running as part of the attribute run? Why isn't this part of manifest?

So first, the manifest of the two AAs might not run. Second, there is a dependency between the two AAs and the order of the manifest call is undetermined.

Reply:
gentle ping